laspy/utils.py

```diff
@@ -12,3 +15,50 @@ def encode_to_null_terminated(string: str, codec: str = "utf-8") -> bytes:
     if b[-1] != 0:
         b += b"\0"
     return b
+
+
+def read_string(
+    stream: BinaryIO, length: int, encoding: str = "ascii "
+) -> Union[str, bytes]:
+    """
+    Reads `length` bytes from the stream, and tries to decode it.
+    If the decoding succeeds, returns the `str`. Otherwise the raw bytes
+    are returned.
+    """
+    raw_string = stream.read(length)
+    first_null_byte_pos = raw_string.find(b"\0")
+    if first_null_byte_pos >= 0:
+        raw_string = raw_string[:first_null_byte_pos]
+
+    try:
+        return raw_string.decode(encoding)
+    except UnicodeDecodeError:
+        return raw_string
+
+
+def write_string(
+    stream: BinaryIO,
+    string: Union[str, bytes],
+    max_length: int,
+    encoding: str = "ascii",
+    encoding_errors: str = "strict",
+) -> bool:
+    """
+    Writes the string or bytes to the stream.
+
+    It will always write `max_length` bytes to the stream,
+    so the input data may be null padded or truncated.
+    """
+    if isinstance(string, str):
+        raw_bytes = string.encode(encoding, errors=encoding_errors)
+    else:
+        # check that the bytes are valid for the given encoding
+        _ = string.decode(encoding, errors=encoding_errors)
+        raw_bytes = string
+
+    if len(raw_bytes) > max_length:
+        stream.write(raw_bytes[:max_length])
+        return True
+    else:
+        stream.write(raw_bytes.ljust(max_length, b"\0"))
```

Comment:
Maybe add a comment here to tell people that these fields require zero-padding if they are shorter than the max_length.

Reply:
You mean in the user documentation ?

Reply:
The code looks weird to someone who is not familiar with the LAS specification and its notion of string fields in headers that are zero-padded. Someone making a patch might drop that because it looks like it doesn't do anything useful without a comment in the code.